src/sagas/background-refresh.js

```diff
@@ -34,28 +34,27 @@ import { fetchIsoFiles } from './storageDomains'
  * This should be done at the time of navigation to the page, typically by the page router.
  */
 function* changePage (action) {
-  yield put(Actions.stopSchedulerFixedDelay())
   yield put(Actions.setCurrentPage(action.payload))
-  yield put(Actions.refresh({
-    onNavigation: true,
-    shallowFetch: true,
-  }))
-  yield put(Actions.startSchedulerFixedDelay())
+  yield put(Actions.startSchedulerFixedDelay({ pageRouterRefresh: true, targetPage: action.payload, startDelayInSeconds: 0 }))
+}
+
+function* refreshManually () {
+  const currentPage = yield select(state => state.config.get('currentPage'))
+  yield put(Actions.startSchedulerFixedDelay({ manualRefresh: true, targetPage: currentPage, startDelayInSeconds: 0 }))
 }
 
 /**
  * Invoke the correct refresh function based on the app's current page type.
  */
-function* refreshData (action) {
-  const currentPage = yield select(state => state.config.get('currentPage'))
+function* refreshData ({ payload: { targetPage, ...otherPayload } }) {
   const refreshType =
-    currentPage.type === C.NO_REFRESH_TYPE ? null
-      : currentPage.type === undefined ? C.LIST_PAGE_TYPE
-        : currentPage.type
+    targetPage.type === C.NO_REFRESH_TYPE ? null
+      : targetPage.type === undefined ? C.LIST_PAGE_TYPE
+        : targetPage.type
 
-  console.info('refreshData() 🡒', 'refreshType:', refreshType, 'currentPage:', currentPage, 'payload:', action.payload)
+  console.info(`refreshData() 🡒 refreshType: ${refreshType} currentPage: ${targetPage} payload: ${otherPayload}`)
```

Comment:
Either swap back to the separated arguments or use `JSON.stringify()` to output the values.  As-is the output will always be:

```
refreshData() 🡒 refreshType: listPage currentPage: [object Object] payload: [object Object]
```

Reply:
@sjd78  I was not aware of this difference! Good to know!